Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPrinting.mm

```diff
@@ -34,13 +34,32 @@
 
 typedef void (^CallCompletionBlock)();
 
-@interface PrintUIDelegate : NSObject <WKUIDelegate>
+@interface WKPrintPageBordersWebView: TestWKWebView
+@end
+
+@implementation WKPrintPageBordersWebView {
+    bool _done;
+}
+
+- (void)drawPageBorderWithSize:(NSSize)borderSize
+{
+    _done = true;
+}
+
+- (void)_waitUntilPrintingDone
+{
+    TestWebKitAPI::Util::run(&_done);
+}
+
+@end
+
+@interface PrintWithDelayedCompletionUIDelegate : NSObject <WKUIDelegate>
 
 - (void)waitForPagination;
 
 @end
 
-@implementation PrintUIDelegate {
+@implementation PrintWithDelayedCompletionUIDelegate {
```

Comment:
Maybe `PrintWithSimulatedPageComputationUIDelegate`? Printing is async, so "delayed completion" isn't really specific to what this class does.

Reply:
this is the current class that was there for the previous test, just renamed (the new name is based on the existing test name). Should I still rename it to `PrintWithSimulatedPageComputationUIDelegate`?

Reply:
I think so, it better distinguishes the two delegates: one that actually presents a print panel (the new one you added), the other that tests an internal method (this existing class).

Reply:
Could also consider one class, with an initializer to choose between the two cases, but not too important since this is testing only code.